props/saber_BC_buttons.h

```diff
@@ -477,738 +1483,1218 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnOnHelper();
+      }
+    }
+#ifdef BC_DUAL_BLADES
+    if (mute_mainBlade_timer_.isTimerExpired()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_MAIN_BLADE_SET);
+        PVLOG_NORMAL << "** Main Blade Turned ON Muted\n";
+      }
+    }
+    if (mute_secondBlade_timer_.isTimerExpired()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_SECOND_BLADE_SET);
+        PVLOG_NORMAL << "** Second Blade Turned ON Muted\n";
+      }
+    }
+#endif
+    // Allow beep to be heard.
+    if (scroll_presets_timer_.isTimerExpired() && scroll_presets_) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
+    if (twist_timer_.isTimerExpired()) {
+      DoSavedTwist();
+    }
   }  // Loop()
 
 #ifdef SPEAK_BLADE_ID
   void SpeakBladeID(float id) override {
     if (&SFX_mnum) {
       sound_library_.SayNumber(id, SAY_WHOLE);
     } else {
-      PVLOG_NORMAL << "No mnum.wav number prompts found.\n";
+      PVLOG_NORMAL << "** No mnum.wav number prompts found.\n";
       beeper.Beep(0.25, 2000.0);
     }
   }
 #endif
 
-// Revert colorchange witout saving (reset to Variation == 0)
-  void ResetColorChangeMode() {
-    if (!current_style()) return;
-      PVLOG_NORMAL << "Reset Color Variation\n";
-      SetVariation(0);
-      PVLOG_NORMAL << "Color change mode done, variation = " << SaberBase::GetCurrentVariation() << "\n";
-      SaberBase::SetColorChangeMode(SaberBase::COLOR_CHANGE_MODE_NONE);
-  }
-
-// Volume Menu
-    void VolumeMenu() {
-    if (!mode_volume_) {
-      current_menu_angle_ = fusor.angle2();
-      mode_volume_ = true;
-      if (SFX_vmbegin) {
-        sound_library_.SayEnterVolumeMenu();
-      } else {
-        beeper.Beep(0.1, 1000);
-        beeper.Beep(0.1, 2000);
-        beeper.Beep(0.1, 3000);
-      }
-      PVLOG_NORMAL << "Enter Volume Menu\n";
-      SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
-    } else {
-      mode_volume_ = false;
-      if (SFX_vmend) {
-        sound_library_.SayVolumeMenuEnd();
-      } else {
-        beeper.Beep(0.1, 3000);
-        beeper.Beep(0.1, 2000);
-        beeper.Beep(0.1, 1000);
-      }
-      PVLOG_NORMAL << "Exit Volume Menu\n";
-    }
-  }
-
-  void VolumeUp() {
-    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
-    current_menu_angle_ = fusor.angle2();
-    if (dynamic_mixer.get_volume() < VOLUME) {
-      dynamic_mixer.set_volume(std::min<int>(VOLUME + VOLUME * 0.1,
-        dynamic_mixer.get_volume() + VOLUME * 0.10));
-      if (SFX_volup) {
-        hybrid_font.PlayPolyphonic(&SFX_volup);
-      } else {
-        beeper.Beep(0.10, 2000);
-        beeper.Beep(0.20, 2500);
-      }
-      PVLOG_NORMAL << "Volume Up - Current Volume: ";
-      PVLOG_NORMAL << dynamic_mixer.get_volume() << "\n";
-    } else {
-      QuickMaxVolume();
+  void DetectMenuTurn() {
```

Comment:
It would make a lot more sense to me if this function updated current_twist_angle_

Reply:
current_twist_angle_ is updated eventually in DoSavedTwist().
This whole twist thing seems kludgy to me, but it's working.
I have yet to try just implementing your other suggestion 
"add the code that clears the strokes array in the regular twist code instead, that way, we can just add some code that checks if there is a single twist in the strokes array, and if it’s been long enough and no new twist has started. Once that happens, we can emit an event and clear the array. The required max time between the strokes is currently 200ms, so we don’t need to wait any longer than that.

PS: the easiest way to clear the array is probably to add a “UNKNOWN_GESTURE” to the StrokeType enum, and then just set strokes[NELEM(strokes)-1].type to UNKNOWN_GESTURE."

Reply:
> current_twist_angle_ is updated eventually in DoSavedTwist().

True, but it is much harder to reason about when and how current_twist_angle_ gets updated.
It would make much more sense if this function handled that part.

The kludgy part is having two different twisty systems IMHO. It would be easier to read and understand how it works if there was only one and it did all the things you wanted it to do...